### PR TITLE
feat: add CI/CD blueprint

### PR DESCRIPTION
**Pull request by wollefitz**
Part of https://github.com/camunda/web-modeler/issues/8961

### CODE REVIEW

**Comment by wollefitz** [comment]
@HanselIdes Could you have a look at the readme if it is easy to understand and additionally the process itself? 

A couple of things to take into account:
* At the moment, it is using collapsed subprocesses so it's not compatible with Zeebe < 8.4 - I can change this if we want to target these cluster versions.
* Under which name to we want to offer this process? I went with "CI/CD Pipeline" for now as you mentioned that "PDLC" is not what users might be used to

File: solutions/ci-cd/README.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+During the process, a pull request is created in the target repository, which, when merged, triggers the deployment to the development environment.
+For deploying to the stage and production environments, custom pipeline runs have to be triggered.
+
+## How to use
+
+### Set up Camunda 8 cluster
+
+#### SaaS
+
+1. In [Console](https://console.cloud.camunda.io), navigate to the "Organization" > "Administration API".
```
Comment (HanselIdes) [inline solutions/ci-cd/README.md:20]: FYI: My first thought when reading this is the cost of maintenance. If you can link to docs page, you will not have to maintain it (although the user experience might not be so good)

I don't have a recommendation on which path to follow though

File: solutions/ci-cd/README.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    * Scopes: `api`, `write_repository`
+    * Roles: `Developer`
+2. Save the token as a new connector secret `GITLAB_PAT` within your cluster.
+
+### Run the blueprint
+
+You can either [start the blueprint via Camunda Tasklist](https://docs.camunda.io/docs/next/components/modeler/web-modeler/run-or-publish-your-process/#publish-to-tasklist) or via running it [directly in Web Modeler](https://docs.camunda.io/docs/next/components/modeler/web-modeler/run-or-publish-your-process/#run-manually-from-modeler).
+Additionally, on Camunda 8 SaaS, you can use[ the public link of the blueprint's start event](https://docs.camunda.io/docs/next/components/modeler/web-modeler/run-or-publish-your-process/#publish-via-a-public-form) to start a new instance of the blueprint.
+
+After starting a process instance, it will create a pull request with the contents of the target folder in the target repository.
```
Comment (HanselIdes) [inline solutions/ci-cd/README.md:100]: If this isn't a dropdown, then the user should get instructions on how to find the folder ID

File: solutions/ci-cd/README.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+For deploying to the stage environment, create a custom pipeline run via "Build > Pipelines > Run pipeline" and additionally pass a variable `ENVIRONMENT` with the value `stage` to the pipeline run:
+
+![Deploy to staging environment](./readme-screenshots/gitlab-custom-pipeline.png)
+
+For deploying to the production environment, pass a variable `ENVIRONMENT` with the value `prod` to the pipeline run.
+
+## Known Limitations
+
+* Concurrent runs of the blueprint are not supported. This means that if you start the blueprint while another instance is still running, deployment information might be mixed between the two instances.
```
Comment (HanselIdes) [inline solutions/ci-cd/README.md:115]: We should be very clear that this is fully customizable, so this is a good place to discuss other limitations you've run into
For example, it's a great place to mention GitHub's limitations around base64 encoding

**Review by HanselIdes — APPROVED** [review]
@wollefitz I'm looking forward to sharing this template with our GitLab-using customers. I added a few non-blocking comments
In addition, there should be a docs PR that advertises this (ideally with an animated visual showing the user experience)

File: solutions/ci-cd/README.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
+# CI/CD Pipeline
+**Technical level:** Complex
+<br>
+**Industry:** Any
+<br>
+**Use Cases:** Modeler Building Blocks
+
+## Summary
+
```
Comment (HanselIdes) [inline solutions/ci-cd/README.md:9]:
```suggestion

Camunda can also be used to orchestrate your process development lifecycle (PDLC), covering all steps from change reviews to release activities.
```

File: solutions/ci-cd/README.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+This blueprint demonstrates a CI/CD pipeline for deploying a file to different environments using GitLab and Camunda Platform 8.
```
Comment (HanselIdes) [inline solutions/ci-cd/README.md:10]:
```suggestion
This blueprint demonstrates a customizable CI/CD pipeline for deploying a file to different environments using GitLab and Camunda Platform 8.
```

File: solutions/ci-cd/CI-CD Pipeline.bpmn [hunk 6]
```diff
@@ -0,0 +1,1579 @@
...
+  "id": "gitlabStartForm",
+  "components": [
+    {
+      "label": "Folder ID",
+      "type": "textfield",
+      "layout": {
+        "row": "Row_0gch4mq",
+        "columns": null
+      },
+      "id": "Field_1ng3kqk",
```
Comment (HanselIdes) [inline solutions/ci-cd/CI-CD Pipeline.bpmn:22]: Readable IDs would be nice to have, especially when they might appear in error messages

File: solutions/ci-cd/CI-CD Pipeline.bpmn [hunk 7]
```diff
@@ -0,0 +1,1579 @@
...
+        "row": "Row_1kqo3ge",
+        "columns": null
+      },
+      "id": "Field_1jvxpgn",
+      "key": "qaSuccessfulCheck",
+      "defaultValue": true
+    }
+  ],
+  "type": "default"
+}</zeebe:userTaskForm>
```
Comment (HanselIdes) [inline solutions/ci-cd/CI-CD Pipeline.bpmn:93]: I recommend adding Documentation as the primary way that a user understands "what is this and how does it work?" instead of just relying on the readme

**Review by HanselIdes — CHANGES_REQUESTED** [review]
I also answered your questions and reviewed the BPMN diagram.
The main thing I'd like to see changed is the addition of documentation. With the new design, it should be a more effective way of explaining the key concepts in case the user wants to modify the process

File: solutions/ci-cd/README.md (hunk 2 above)
Comment (wollefitz) [reply solutions/ci-cd/README.md:100]: Added some information plus a screenshot

File: solutions/ci-cd/CI-CD Pipeline.bpmn (hunk 7 above)
Comment (wollefitz) [reply solutions/ci-cd/CI-CD Pipeline.bpmn:131]: @HanselIdes I added a short introduction to both forms to clarify the usage of the blueprint. Do you think that's enough or do you think "in-diagram" documentation should be more verbose in general?

Comment (wollefitz) [reply solutions/ci-cd/CI-CD Pipeline.bpmn:131]: Great point! Added this limitation and another sentence to the introduction: "The blueprint is designed to be used with GitLab, but can be adapted to work with other CI/CD tools."

File: solutions/ci-cd/README.md (hunk 3 above)
Comment (wollefitz) [reply solutions/ci-cd/README.md]: Great point! Added this limitation and another sentence to the introduction: "The blueprint is designed to be used with GitLab, but can be adapted to work with other CI/CD tools."

**Comment by wollefitz** [comment]
FYI: @HanselIdes Till was so kind to review the blueprint in a separate PR (see https://github.com/camunda/camunda-platform-tutorials/pull/61) which is merged to this branch now. Could you have another look at the changes?

File: solutions/ci-cd/README.md [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+The blueprint is designed to be used with GitLab, but can be adapted to work with other CI/CD tools.
+
+## How to use
+
+### Set up Camunda 8 cluster
+
+#### SaaS
+
+1. Create an API client [as described in the documentation](https://docs.camunda.io/docs/next/apis-tools/web-modeler-api/authentication/?authentication=saas) (with at least "Get" and "Create" scopes for "Web Modeler API").
+2Camunda can also be used to orchestrate your process development lifecycle (PDLC), covering all steps from change reviews to release activities.. Click on "Create" and create a connector secret `WEB_MODELER_CLIENT_ID` and `WEB_MODELER_CLIENT_SECRET` in the cluster you want to use the blueprint in and set the values to the "Client ID" and "Client Secret" respectively.
```
Comment (HanselIdes) [inline solutions/ci-cd/README.md:23]:
```suggestion
2. Camunda can also be used to orchestrate your process development lifecycle (PDLC), covering all steps from change reviews to release activities.. Click on "Create" and create a connector secret `WEB_MODELER_CLIENT_ID` and `WEB_MODELER_CLIENT_SECRET` in the cluster you want to use the blueprint in and set the values to the "Client ID" and "Client Secret" respectively.
```

File: solutions/ci-cd/README.md [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+- **Target Base Branch:** In the course of the blueprint, a new branch will be created in the target repository.
+  This field specifies the base branch for the new branch.
+- **Target GitLab Base URL:** The base URL of the GitLab instance. Defaults to `https://gitlab.com`.
+- **Target GitLab Project ID:** The ID of the target project in GitLab.
+  You can find the ID in the URL of the project in GitLab:
+  ![Project ID](./readme-screenshots/project-id.png)
+
+After starting a process instance, it will create a pull request with the contents of the target folder in the target repository.
+When you merge the pull request, the folder contents will be deployed to the development environment via [the `deploy_to_dev` job within the pipeline](./gitlab-pipeline/.gitlab-ci.yml).
+To check the status of the process, you can always use [Camunda Operate](https://docs.camunda.io/docs/next/components/operate/operate-introduction/) to have a look at the status of the process.
```
Comment (HanselIdes) [inline solutions/ci-cd/README.md:115]:
```suggestion
To check the status of the process, you can always use [Camunda Operate](https://docs.camunda.io/docs/next/components/operate/operate-introduction/) to track the status of the process.
```

File: solutions/ci-cd/README.md [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+Operate should indicate with blue colored sequence flows the where to process went and mark the current activity with a green circle:
```
Comment (HanselIdes) [inline solutions/ci-cd/README.md:116]:
```suggestion
Operate should indicate with blue colored sequence flows where the process went and mark the current activity with a green circle:
```

File: solutions/ci-cd/README.md [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+**Technical level:** Complex
+<br>
+**Industry:** Any
+<br>
+**Use Cases:** Modeler Building Blocks
+
+## Summary
+
+Camunda can also be used to orchestrate your process development life cycle (PDLC), covering all steps from change reviews to release activities.
+This blueprint demonstrates a customizable CI/CD pipeline for deploying a file to different environments using GitLab and Camunda Platform 8.
```
Comment (HanselIdes) [inline solutions/ci-cd/README.md:11]:
```suggestion
This blueprint demonstrates a customizable CI/CD pipeline for deploying a file to different environments using GitLab and Camunda Platform 8. To learn more about the topic, review [Integrate Web Modeler into CI/CD](https://docs.camunda.io/docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd/)
```

**Review by HanselIdes — APPROVED** [review]
I contributed some process documentation and recommended some grammar and spelling changes. You don't need to re-request my review.

File: solutions/ci-cd/CI-CD Pipeline.bpmn (hunk 7 above)
Comment (HanselIdes) [reply solutions/ci-cd/CI-CD Pipeline.bpmn:134]: I think the in-diagram documentation should be more verbose and explain what's happening in plain English. I shared some examples to the top level of process.
